sources/common/infoschema.go

```diff
@@ -133,6 +139,141 @@ func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
 	}
 }
 
+func ProcessDataWithDataproc(conv *internal.Conv, infoSchema InfoSchema, dataprocConfig map[string]string) error {
+
+	orderTableNamesByID := ddl.GetSortedTableIdsBySpName(conv.SpSchema)
+	numberOfTables := int64(len(orderTableNamesByID))
+
+	if !conv.Audit.DryRun {
+		conv.Audit.Progress = *internal.NewProgress(numberOfTables, "Writing data to Spanner via Dataproc", internal.Verbose(), false, int(internal.DataWriteInProgress))
+	}
+
+	progressCtr := 0
+
+	for _, spannerTableID := range orderTableNamesByID {
+
+		srcTable := conv.SrcSchema[spannerTableID].Name
+
+		srcSchema := conv.SrcSchema[spannerTableID]
+
+		primaryKeys, _, _ := infoSchema.GetConstraints(conv, SchemaAndName{Name: srcTable, Schema: srcSchema.Schema})
+
+		id, err := TriggerDataprocTemplate(srcTable, srcSchema.Schema, strings.Join(primaryKeys, ","), dataprocConfig)
+		if err != nil {
+			return err
+		}
+		if conv.DataFlush != nil {
+			conv.DataFlush()
+		}
+
+		if !conv.Audit.DryRun {
+			progressCtr++
+			conv.Audit.Progress.MaybeReport(int64(progressCtr))
+		}
+
+		//TODO: eenclona@ will remove hardcoded us-central1 to be parametarized
+		url := fmt.Sprintf("https://pantheon.corp.google.com/dataproc/batches/us-central1/%s", id)
+		conv.Audit.DataprocStats.DataprocJobUrls = append(conv.Audit.DataprocStats.DataprocJobUrls, url)
+		conv.Audit.DataprocStats.DataprocJobIds = append(conv.Audit.DataprocStats.DataprocJobIds, id)
+
+	}
+
+	return nil
+}
+
+// Function to trigger dataproc template
+func TriggerDataprocTemplate(srcTable string, srcSchema string, primaryKeys string, dataprocConfig map[string]string) (string, error) {
+	ctx := context.Background()
+
+	println("Triggering Dataproc template for " + srcSchema + "." + srcTable)
+
+	// Extract location from subnet
+	subnet := dataprocConfig["subnet"]
+	region_string := subnet[0:strings.Index(subnet, "/subnetworks")]
```

Comment:
This is error prone, region should be passed as its own parameter IMO

Reply:
For now, just added validators in front-end to validate subnet uri before sending it to backend.
Exploring other options for future to simplify it further